modules/textlocalwrapper.py

```diff
@@ -0,0 +1,111 @@
+import json
+import re
+import string
+
+from six import u, unichr
+from six.moves.urllib import request, parse
+from datetime import timedelta
+
+from cshsms.settings import TEXTLOCAL_API, TEXTLOCAL_PRIMARY_ID
+from modules.date_helper import datetime_from_date_string
+
+
+class TextLocal(object):
+    def __init__(self, apikey, primary_id):
+        self.apikey = apikey
+        self.primary_id = primary_id
+
+
+    def get_all_inboxes(self):
+        params = {'apikey': self.apikey}
+        inboxes_url = 'https://api.textlocal.in/get_inboxes/?'
+        return self.get_url_response(request_url=inboxes_url, params=params)
+
+
+    def get_primary_inbox(self):
+        params = {'apikey': self.apikey, 'inbox_id': self.primary_id}
+        messages_url = 'https://api.textlocal.in/get_messages/?'
+        return self.get_url_response(request_url=messages_url, params=params)
+
+
+    def get_url_response(self, request_url, params):
+        f = request.urlopen(request_url + parse.urlencode(params))
+        return json.loads(f.read().decode('latin1'))
+
+
+    def get_primary_inbox_messages(self):
+        return self.get_primary_inbox()['messages']
+
+
+    def combine_messages(self, messages):
+        # TextLocal will split messages that are too long into multiple messages.
+        # This can (hopefully) be detected by looking for messages sent less than ten
+        # seconds apart where the first message in the pair has a length of >250 chars.
+        # ...Also, of course, the order of the message can't be guaranteed... :(
+        # TODO: This is a very terrible loop :(
+        combined_messages = []
+        i = 0
+        while i < len(messages):
+            if i == len(messages) - 1:
+                combined_messages.append(messages[i])
+                i += 1
+            else:
+                first_message_date = datetime_from_date_string(messages[i]['date'], "%Y-%m-%d %H:%M:%S")
+                second_message_date = datetime_from_date_string(messages[i + 1]['date'], "%Y-%m-%d %H:%M:%S")
+                ten_seconds = timedelta(0, 10)
+                if second_message_date - first_message_date < ten_seconds:
```

Comment:
This combines without regard to the phone number of the message. This means messages from two different numbers would combine. And the way this is set up, the final combined message would be assigned the phone number of the final message added to the combined message, so you wouldn't know this happened.

Reply:
Ah yeah, I should take care to make sure that `number` is the same before combining.

Reply:
This is done now.